infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactory.java

```diff
@@ -86,12 +95,17 @@ public OpenShiftProjectFactory(
    * @throws InfrastructureException if any exception occurs during project preparing
    */
   public OpenShiftProject create(String workspaceId) throws InfrastructureException {
-    final String projectName =
-        evalNamespaceName(workspaceId, EnvironmentContext.getCurrent().getSubject());
+    final String projectName;
+    try {
+      projectName = evalNamespaceName(workspaceId, EnvironmentContext.getCurrent().getSubject());
+    } catch (NotFoundException | ServerException | ConflictException | ValidationException e) {
+      throw new InfrastructureException(
+          format("Failed to evaluate the project name to use for workspace %s.", workspaceId), e);
+    }
     OpenShiftProject osProject = doCreateProject(workspaceId, projectName);
     osProject.prepare();
 
-    if (!isPredefined() && !isNullOrEmpty(getServiceAccountName())) {
+    if (isCreatingNamespace(workspaceId) && !isNullOrEmpty(getServiceAccountName())) {
```

Comment:
```suggestion
    if (!isNullOrEmpty(getServiceAccountName() && isCreatingNamespace(workspaceId))) {
```
`isCreatingNamespace` method has quite heavy logic but in fact you don't need to evaluate it if serviceAccountName is null or empty. Since workspace SA is rarely empty or null it's minor but still optimization

Reply:
Hence the order of the checks in this condition. The `isCreatingNamespace` is not called if there is no service account configured.